src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -7104,29 +7095,30 @@ void wm_vuldet_run_scan(wm_vuldet_t *vuldet) {
         return;
     }
 
-    if (wm_vuldet_set_agents_info(vuldet->node_name, &vuldet->agents_software, vuldet->updates)) {
+    if (wm_vuldet_collect_agents_to_scan(vuldet)) {
         mterror(WM_VULNDETECTOR_LOGTAG, VU_NO_AGENT_ERROR);
-    } else {
-        if (wm_vuldet_check_agent_vulnerabilities(vuldet->agents_software, &vuldet->flags, vuldet->min_full_scan_interval)) {
+    }
+    else {
+        if (wm_vuldet_check_agent_vulnerabilities(vuldet)) {
             mterror(WM_VULNDETECTOR_LOGTAG, VU_AG_CHECK_ERR);
         } else {
             mtinfo(WM_VULNDETECTOR_LOGTAG, VU_END_SCAN);
         }
     }
 
-    agent_software *agent;
-    for (agent = vuldet->agents_software; agent;) {
-        agent_software *agent_aux = agent->next;
+    scan_agent *agent;
+    for (agent = vuldet->scan_agents; agent;) {
+        scan_agent *agent_aux = agent->next;
 
-        wm_vuldet_free_agent_software(agent);
+        wm_vuldet_free_scan_agent(agent);
         if (agent_aux) {
             agent = agent_aux;
         } else {
             break;
         }
     }
 
-    vuldet->agents_software = NULL;
+    vuldet->scan_agents = NULL;
     vuldet->last_detection = time(NULL);
```

Comment:
We were talking with the team that the name `last_detection` could be modified.

Reply:
Now `last_detaction` was renamed to `last_scan` and `detaction_interval` to `scan_interval`